fields/types/numberarray/NumberArrayField.js

```diff
@@ -11,7 +11,7 @@ module.exports = Field.create({
 	mixins: [ArrayFieldMixin],
 
 	cleanInput (input) {
-		return input.replace(/[^\d]/g, '');
+		return input.replace(/[^\d\.]/g, '');
```

Comment:
Thanks for taking a look at this issue! Unfortunately the fix isn't as simple as allowing `.` in the input, as that allows for invalid Number values like `3.4.5` or `..`.

I suggest comparing with the validation for the [`Number` field type](https://github.com/keystonejs/keystone/blob/master/fields/types/number/NumberField.js), which validates appropriately.

- Some allowable values to test: 12.34, -12.34, 0
- Some invalid values: A, 12.34.56

Once you have the code change working, please add or update the end-to-end (e2e) functional test coverage. As per the pull request template, please also list the specific browser versions that were tested.

For e2e testing Keystone 4 uses [Nightwatch.js](http://nightwatchjs.org/). See the [`test/e2e`](https://github.com/keystonejs/keystone/tree/master/test/e2e) directory in the Keystone repo for information. The e2e tests run automatically on every PR or commit, but you can run them much faster locally before pushing a PR.

Regards,
Stennie